gallery/plot_transforms_v2_e2e.py

```diff
@@ -0,0 +1,147 @@
+"""
+==================================================
+transforms v2: End-to-end object detection example
+==================================================
+"""
+
+import pathlib
+from collections import defaultdict
+
+import PIL.Image
+
+import torch
+import torch.utils.data
+
+import torchvision
+
+
+def show(sample):
+    import matplotlib.pyplot as plt
+
+    from torchvision.transforms.v2 import functional as F
+    from torchvision.utils import draw_bounding_boxes
+
+    image, target = sample
+    if isinstance(image, PIL.Image.Image):
+        image = F.to_image_tensor(image)
+    image = F.convert_dtype(image, torch.uint8)
+    annotated_image = draw_bounding_boxes(image, target["boxes"], colors="yellow", width=3)
+
+    fig, ax = plt.subplots()
+    ax.imshow(annotated_image.permute(1, 2, 0).numpy())
+    ax.set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
+    fig.tight_layout()
+
+    fig.show()
+
+
+########################################################################################################################
+# To use ``transforms.v2``, you can use the stable datasets and models and just have import transforms differently
+
+# We are using BETA APIs, so we deactivate the associated warning, thereby acknowledging that
+# some APIs may slightly change in the future
+torchvision.disable_beta_transforms_warning()
+
+from torchvision import models, datasets
+import torchvision.transforms.v2 as transforms
+
+
+########################################################################################################################
+# We start off by loading the :class:`~torchvision.datasets.CocoDetection` dataset to have a look at the status quo.
+
+
+def load_example_coco_detection_dataset(**kwargs):
+    # This loads fake data for illustration purposes of this example. In practice, you'll have
+    # to replace this with the proper data
+    root = pathlib.Path("assets") / "coco"
+    return datasets.CocoDetection(str(root / "images"), str(root / "instances.json"), **kwargs)
+
+
+dataset = load_example_coco_detection_dataset()
+
+sample = dataset[0]
+image, target = sample
+print(type(image))
+print(type(target))
+print(type(target[0]))
+print(list(target[0].keys()))
+
+
+########################################################################################################################
+# The dataset returns a two-tuple with the first item being a :class:`PIL.Image.Image` and second one a list of
+# dictionaries, which each containing the annotations for a single object instance. As is, this format is not compatible
+# with the transforms v2, but also the models. To overcome that, we provide the
+# :func:`~datasets.wrap_dataset_for_transforms_v2` function. For :class:`~datasets.CocoDetection` this changes the
```

Comment:
```suggestion
# :func:`~torchvision.datapoints.wrap_dataset_for_transforms_v2` function. For :class:`~torchvision.datasets.CocoDetection` this changes the
```

Reply:
You mean 

```suggestion
# :func:`~torchvision.datasets.wrap_dataset_for_transforms_v2` function. For :class:`~torchvision.datasets.CocoDetection` this changes the
```

right? Recall that the wrapper is now located in the dataset namespace.

Reply:
I went for my suggestion, although for now the link is broken anyway, since the wrapper function is not yet documented.